modules/dcache/src/main/java/org/dcache/services/ssh2/Ssh2Admin.java

```diff
@@ -318,6 +327,66 @@ public boolean authenticate(String userName, PublicKey key,
         }
     }
 
+    private class RemoteHostValidator {
```

Comment:
I don't think there's any hard-and-fast rules here -- certainly, having the class doesn't prevent the patch from being accepted.

Just to describe my thinking: normally, a class is present for a reason -- it makes working with the code easier by provides encapsulation (hiding state), abstraction (through polymorphism), or promotes code reuse.  None of these apply here.

Classes also exist to split off functionality -- a class that tries to do too much is then unwieldy: difficult to work with.  So, one could argue that grouping the methods together into class is about trying to keep Ssh2Admin class small.  However, that argument kinda falls flat, because `RemoteHostValidator` is an inner class, so Ssh2Admin is actually slightly larger that if there were no RemoteHostValidator class.

On the flip side, there is a cost to having the class.  Including `RemoteHostValidator` actually makes the code a little harder to read.  Rather than directly invoking a method to test whether the login is acceptable (as a filter in the stream processing), a "helper" method is called (`remoteHostIsPermitted`) that instantiates a class (without any state) and calls the method.  Without this class, the `isValidHost` method could be called directly from the stream.

So, it's ultimately about balance: does the benefits of having the class outweigh the cost?

In the end, it's your call.  I'd say currently, the costs outweigh the benefits.  ... but this is just code: we can always change it later on; for example, should we need to "configure" how host validation works; e.g., with decorator and/or strategy design patterns.